optaplanner-test/src/main/java/org/optaplanner/test/api/change/MockProblemChangeDirector.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.test.api.change;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.solver.change.ProblemChangeDirector;
+
+/**
+ * Use for unit-testing {@link org.optaplanner.core.api.solver.change.ProblemChange}s.
+ */
+public class MockProblemChangeDirector implements ProblemChangeDirector {
```

Comment:
See the usage e.g. here:
https://github.com/kiegroup/optaweb-vehicle-routing/pull/635/files#diff-4b76b19ef893258e0868554c60fc2784cff0a8c1d6fdb2d4de38a3d08f3968e3

Reply:
Just to confirm:
- This does not expose Mockito API in our public API of optaplanner-test? (it should not and it doesn't even seem to use Mockito)
- The code in optaplanner-test is public API. Written in stone. Are we sure we want to bring this class in the public API already? We can also just embed it in optaweb-vehicle-routing.

I am ok with bringing this in the public API of optaplanner-test, but I 'd love a clearer javadoc on the class to tell why and when I 'd use this thing.

Reply:
It's not just a matter of the optaweb-vehicle-routing, as the task-assigning in kogito-apps heavily tests PFCs too.

The downside of the new `ProblemChangeDirector` API is that simple mocking of ScoreDirector is no longer possible, as the code that changes the working solution is not called by the PFC implementation directly. The caller now is the `ProblemChangeDirector`. This makes unit testing impossible without a `ProblemChangeDirector` implementation available for tests.

+1 to enhance the javadoc.